assets/javascripts/discourse-markdown/notifications.js.es6

```diff
@@ -9,43 +9,47 @@ export function setup(helper) {
             'div.p-notification--negative',
             'div.p-notification--important',
             'p.p-notification__response',
-            'span.p-notification__status',
-            'span.p-notification__line'
+            'span.p-notification__status'
         ]
     );
 
     helper.registerPlugin(md=>{
-        md.block.bbcode.ruler.push('note', {
-            tag: 'note',
+        md.block.bbcode.ruler.push("note", {
+            tag: "note",
             replace: function(state, tagInfo, content) {
-                let status = "";
-                let type = "";
-
-                if ('status' in tagInfo.attrs) {
-                    status = '<span class="p-notification__status">'
-                    + tagInfo.attrs.status
-                    + ':</span>'
-                }
-
+                // Build up the notification class from the specified "type"
+                // "p-notification--{type}"
+                let notificationClass = 'p-notification'
                 if (
                     'type' in tagInfo.attrs
                     && ['caution', 'positive', 'negative', 'important'].includes(
                         tagInfo.attrs.type.toLowerCase()
                     )
                 ) {
-                    type = "--" + tagInfo.attrs.type.toLowerCase();
+                    notificationClass += "--" + tagInfo.attrs.type.toLowerCase();
                 }
 
-                let notification_block = state.push('html_raw', '', 0);
+                // Start wrapper elements:
+                // <div class="p-notification"><p class="p-notification__render">
+                state.push('div_open', 'div', 1).attrSet('class', notificationClass);
```

Comment:
I think this lets arbitrary classes be injected, by putting spaces in the type.

Reply:
I don't believe there's any way to bypass the `['caution', 'positive', 'negative', 'important'].includes(                        tagInfo.attrs.type.toLowerCase())` conditional - as in, if we reach this line, the `type` much exactly equal one of those 4 values.

But I could rewrite it as the more verbose:

``` javascript
let type = tagInfo.attrs.type.toLowerCase();
let notificationClass = 'p-notification';

if (type  === 'positive') {
    notificationClass = 'p-notification--positive';
} else if (type === 'negative') {
    notificationClass = 'p-notification--negative';
} else if (type === 'caution') {
    notificationClass = 'p-notification--caution';
} else if (type === 'important') {
    notificationClass = 'p-notification--important';
}
```

If you think it's safer.

Reply:
Er sorry, you're right, missed that it was only in that conditional. This is fine.